onnxruntime/core/providers/webnn/builders/model.cc

```diff
@@ -26,6 +26,11 @@ Model::~Model() {}
 
 Status Model::Predict(const InlinedHashMap<std::string, OnnxTensorData>& inputs,
                       const InlinedHashMap<std::string, OnnxTensorData>& outputs) {
+  // Allocate the MLNamedArrayBufferViews for inputs and outputs at every compute()
+  // because they will be transferred after compute() done.
+  // https://webmachinelearning.github.io/webnn/#api-mlcontext-async-execution
+  AllocateInputOutputBuffers();
```

Comment:
Thanks @huningxin, indeed we can reuse that buffers from compute() results to void additional allocation. I've fixed that in latest comment, PTAL again, thanks!

Reply:
I am not sure how buffer reuse works in WebNN, but we can specify tensor's location is 'cpu'(default) or 'cpu-pinned'(the preserved type for buffer reuse). Please let me know if you think webnn can leverage this interface.

BTW, I think this buffer reuse is not a part of this PR.

Reply:
This comment is just to avoid unnecessary input and output buffers allocation for every compute call. For a subgraph, WebNN EP can pre-allocate the input and output buffers at session creation time and reuse those buffers for compute. I think @Honry 's latest change reflects that.

> we can specify tensor's location is 'cpu'(default) or 'cpu-pinned'(the preserved type for buffer reuse). Please let me know if you think webnn can leverage this interface.

This sounds interesting. Could you please share some pointers so we can learn more?